checker/src/main/java/org/checkerframework/checker/signedness/jdk.astub

```diff
@@ -3268,6 +3268,12 @@ class PrintStream {
 class RandomAccessFile {
     public void write(@PolySigned byte b[]);
     public void write(@PolySigned byte b[], int off, int len);
+    public int read(@PolySigned byte[] b) throws IOException;
+    public int read(@PolySigned byte b[], int off, int len) throws IOException;
+    public native @Constant long length() throws IOException;
```

Comment:
I wanted it to be interpreted both as unsigned or signed that's why I did that. But signedness checker has a feature that it can interpret it as both signed and unsigned if it is within signed positive range so it would be good to annotate it with ```@Unsigned```. I have corrected it.

Reply:
Why do you want to interpret it both as signed or unsigned?
Presumably because it is definitely non-negative (the MSB is zero).
It would be helpful to write such a comment for clarification.
I would not write it as `@Unsigned`, because that would restrict its uses.

Reply:
I wanted it to interpreted as both because as you said as ```@Unsigned```, it would restrict its uses.
And in my case study, it was giving an error but by annotating the method with the annotation it resolved the issue.
I think for now we should not annotate this method. A proper study is needed or maybe a new annotation to solve this issue.

Reply:
You are right that we can interpret signed positive numbers as either signed or unsigned.
The Signedness Checker [now](https://github.com/typetools/checker-framework/commit/e412a5f9a55937edeea04ebfddd0984786a0f1e9?w=1) behaves that way for all values -- before it only had that behavior for compile-time constants.

With the new feature, no Signedness annotation is needed -- the `@NonNegative` annotation on `length()`, which ought to already appear in the JDK, is sufficient, and I think is more clear.

I also opened [an issue](https://github.com/typetools/checker-framework/issues/2457) to rename `@Constant` to `@SignednessEither`.  It is misleading to call the annotation `@Constant` when it applies more broadly than that.

Thanks for bringing up this point!